docs/changes/newsfragments/7012.improved

```diff
@@ -0,0 +1 @@
+Change snapshot to use the register_name of the parameter instead of short_name as key in the parameters snapshot.
```

Comment:
```suggestion
Use the register_name of the parameter instead of short_name as key in the parameters snapshot in datasets (not the station snapshot).
```